### DISCUSSION TITLE
Non-overlapping ranges for event and channel IDs

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1347] [Q&A] [answered]
I am intending to enforce in my deployment non-overlapping ranges for channel and event IDs, so that these IDs can mimic being taken from the same ID pool. This appears to work OK so far, but I wondered if anyone has had problems with this before.

For example, given a base component ID of X, then event IDs run from X to X+49 and channel IDs from X+50 to X+99.

**Comment by timcanham** [comment, marked as answer]
This should be fine. You just have to keep track of the overall ID range and make sure it doesn't overlap another component.
